Compatibility-2.0.md

```diff
@@ -26,3 +26,304 @@ Embedded Mode also might use a decorated Cache using a Transcoder. For example t
 We must ensure that Consistent-Hash-aware clients knowledge of the hashing of keys is not affected by transcoding.
 
 Transcoders should be deployable
+
+
+# Encoders, Transcoders and indexable Json 
+
+[ISPN-7896](https://issues.jboss.org/browse/ISPN-7896) introduced the Encoders, adding capability to the caches for on-demand data conversion instead of relying on implicit conversions based on a combination of flags and interceptors. 
+
+This section will expand on the Encoders usage and will introduce the complementary Transcoders, used to convert between different encodings.
+
+## About Encoders
+
+An Encoder is essentially a two-way conversion between the storage format in a cache and the read/write format. The ```AdvancedCache```  interface has a method to specify an encoder to be used during all operations involving that cache:
```

Comment:
The wire transfers bytes according to the Externalizer, the rw format is the format used on the API border.